src/components/molecules/Modal/Modal.jsx

```diff
@@ -0,0 +1,55 @@
+import './Modal.scss';
+import { useState, useEffect, useCallback } from 'react';
+import PropTypes from 'prop-types';
+import { Icon } from '../../atoms/IconComponent/Icon';
+
+export const Modal = ({ children }) => {
+    const [showModal, setShowModal] = useState(false);
+
+    const keyPress = useCallback(
+        (e) => {
+            if (e.key === 'Escape' && showModal) {
+                setShowModal(false);
+            }
+        },
+        [setShowModal, showModal]
+    );
+
+    const toggleModal = () => {
+        setShowModal((prev) => !prev);
+    };
+
+    useEffect(() => {
+        document.addEventListener('keydown', keyPress);
+        return () => document.removeEventListener('keydown', keyPress);
+    });
+
+    return (
+        <>
+            <Icon
```

Comment:
This icon shouldn't be here, we want to reuse the modal with any button, not the one here